File: 122COM/OldSchoolGame/fighter.cpp
#include <iostream>
#include <stdlib.h>
#include <string>
#include "fighter.hpp" 


using namespace std ;

Fighter::Fighter(string name) : Player (name)
{
    setTimer(2) ;
    setChargingAttack(true) ; 
}

void Fighter::powerAttack(Player &enemy)
{
/* In order for this function to work, we need to 
call it 3 times. Otherwise, we charge up without attacking. */    
    if (getTimer() > 0) 
    {

        if (getTimer() == 2) 
        {
            cout << getTimer() << " turns to charge up." << endl ; 
        }
        else
        {
            cout << getTimer() << " turn to charge up" << endl ; 
        }  // else statement deals with singular (turn) vs plurar (turns)
        timerTick() ; 
    }
    else 
    {
        swapChargingAttack() ;
        resetTimer() ; 

        cout << "You smacked him in the head real hard." << endl ; 
        enemy.takeDamage(70) ; // smacks the enemy in the head real hard 
    }
        
}

void Fighter::expAttack(Player &enemy)
{
    int rng = rand() % 30  ;  // random damage, 5-35 
    int damage = rng + 5 ; 
    enemy.takeDamage(damage) ; 
    if (enemy.getHP() <= 0)  // if the hit kills the enemy, we get extra xp 
    {
        gainEXP(25) ; 
        cout << "You just gained some xp!" << endl ; 
    }
}

void Fighter::setTimer(int value)
{
    this-> timer = value ; 
}

void Fighter::timerTick()
{
    this-> timer -= 1 ; 
}

void Fighter::resetTimer()
{
    this-> timer = 2 ; 
}

void Fighter::setChargingAttack(bool value)
{
    this-> chargingAttack = value ; 
}

void Fighter::swapChargingAttack()
{
    this-> chargingAttack = not chargingAttack ; 
}

int Fighter::getTimer()
{
    return this-> timer ; 
}

File: 122COM/OldSchoolGame/Python_Design/fighter.py
from player import Player,random


class Fighter(Player):

    def __init__(self, name):
        Player.__init__(self,name)
        self.baseDMG += 10
        self.armor += 5
        self.hp += 25
        self.chargingAttack = True
        self.timer = 2
            # START WITH A SWORD

# simple attack deals 25 dmg
# crit strike deals 50 dmg
# power attack deals 70 dmg
# exp attack deals 10-30 dmg
    #if you kill the enemy, gain 25XP

    def powerAttack(self,other):
        if self.timer > 0:
            self.chargingAttack = False
            if self.timer == 2:
                print(str(self.timer) + " turns to charge up.")
            else:
                print(str(self.timer) + " turn to charge up.")
            self.timer -= 1
            
            return
        else:
            self.chargingAttack = True
            self.timer = 2
            print("You smacked him in the head real hard.")
            other.takeDamage(70)

    def expAttack(self, other):
        damage = random.randint(5,35)
        other.takeDamage(damage)
        if other.hp <= 0:
            self.gainEXP(25)
            #extra gold as well once we have it 
